### PR TITLE
Fix #1623 vic-machine delete failed for vSAN

### LINKED ISSUE #1623: vic-machine delete fails with "cannot delete file" for vSAN image store
Very simple to reproduce. Just create a VCH, pull an image and delete the VCH.

```
root@photon-machine [ ~/vic/src/github.com/vmware/vic/bin ]# ./vic-machine-linux create --target=10.118.67.155 --user Administrator@vsphere.local --compute-resource cluster1 --image-datastore vsanDatastore --password ca\$hc0w --external-network=ExternalNetwork --bridge-network=InternalNetwork --name test6
INFO[2016-07-22T07:00:40Z] ### Installing VCH ####                      
INFO[2016-07-22T07:00:40Z] Generating certificate/key pair - private key in ./test6-key.pem 
INFO[2016-07-22T07:00:41Z] Validating supplied configuration            
INFO[2016-07-22T07:00:41Z] Firewall status: ENABLED on "/ha-datacenter/host/cluster1/10.118.67.56" 
INFO[2016-07-22T07:00:41Z] Firewall status: ENABLED on "/ha-datacenter/host/cluster1/10.118.67.57" 
INFO[2016-07-22T07:00:41Z] Firewall status: ENABLED on "/ha-datacenter/host/cluster1/10.118.67.58" 
INFO[2016-07-22T07:00:41Z] Firewall configuration OK on hosts:          
INFO[2016-07-22T07:00:41Z]   "/ha-datacenter/host/cluster1/10.118.67.56" 
INFO[2016-07-22T07:00:41Z]   "/ha-datacenter/host/cluster1/10.118.67.57" 
INFO[2016-07-22T07:00:41Z]   "/ha-datacenter/host/cluster1/10.118.67.58" 
INFO[2016-07-22T07:00:41Z] License check OK on hosts:                   
INFO[2016-07-22T07:00:41Z]   "/ha-datacenter/host/cluster1/10.118.67.56" 
INFO[2016-07-22T07:00:41Z]   "/ha-datacenter/host/cluster1/10.118.67.57" 
INFO[2016-07-22T07:00:41Z]   "/ha-datacenter/host/cluster1/10.118.67.58" 
INFO[2016-07-22T07:00:41Z] DRS check OK on:                             
INFO[2016-07-22T07:00:41Z]   "/ha-datacenter/host/cluster1/Resources"   
INFO[2016-07-22T07:00:43Z] Creating virtual app "test6"                 
INFO[2016-07-22T07:00:43Z] Creating appliance on target                 
INFO[2016-07-22T07:00:43Z] Network role "client" is sharing NIC with "external" 
INFO[2016-07-22T07:00:43Z] Network role "management" is sharing NIC with "external" 
INFO[2016-07-22T07:01:04Z] Uploading images for container               
INFO[2016-07-22T07:01:04Z] 	"bootstrap.iso"                             
INFO[2016-07-22T07:01:04Z] 	"appliance.iso"                             
INFO[2016-07-22T07:01:12Z] Registering VCH as a vSphere extension       
INFO[2016-07-22T07:01:17Z] Waiting for IP information                   
INFO[2016-07-22T07:01:34Z] Waiting for major appliance components to launch 
INFO[2016-07-22T07:01:34Z] Initialization of appliance successful       
INFO[2016-07-22T07:01:34Z]                                              
INFO[2016-07-22T07:01:34Z] Log server:                                  
INFO[2016-07-22T07:01:34Z] https://10.118.67.51:2378                    
INFO[2016-07-22T07:01:34Z]                                              
INFO[2016-07-22T07:01:34Z] DOCKER_HOST=10.118.67.51:2376                
INFO[2016-07-22T07:01:34Z]                                              
INFO[2016-07-22T07:01:34Z] Connect to docker:                           
INFO[2016-07-22T07:01:34Z] docker -H 10.118.67.51:2376 --tls info       
INFO[2016-07-22T07:01:34Z] Installer completed successfully             
root@photon-machine [ ~/vic/src/github.com/vmware/vic/bin ]# docker -H 10.118.67.51:2376 --tls pull busybox
Using default tag: latest
latest: Pulling from library/busybox
a3ed95caeb02: Pull complete 
8ddc19f16526: Pull complete 
Digest: sha256:65ce39ce3eb0997074a460adfb568d0b9f0f6a4392d97b6035630c9d7bf92402
Status: Downloaded newer image for library/busybox:latest
root@photon-machine [ ~/vic/src/github.com/vmware/vic/bin ]# ./vic-machine-linux delete --target=10.118.67.155 --user Administrator@vsphere.local --compute-resource cluster1 --password ca\$hc0w --name test6
INFO[2016-07-22T07:02:02Z] ### Removing VCH ####                        
INFO[2016-07-22T07:02:04Z] Removing VMs                                 
INFO[2016-07-22T07:02:04Z] Removing images                              
ERRO[2016-07-22T07:02:04Z] Operation failed: Cannot delete file [vsanDatastore] VIC/42147648-5adf-788d-6387-6706b233ef82/images - (&{{} %!!(MISSING)s(*types.CannotDeleteFile=&{{{{<nil> []}} [vsanDatastore] VIC/42147648-5adf-788d-6387-6706b233ef82/images}}) Cannot delete file [vsanDatastore] VIC/42147648-5adf-788d-6387-6706b233ef82/images}) 
ERRO[2016-07-22T07:02:04Z] Operation failed: Cannot delete file [vsanDatastore] VIC/42147648-5adf-788d-6387-6706b233ef82 - (&{{} %!!(MISSING)s(*types.CannotDeleteFile=&{{{{<nil> []}} [vsanDatastore] VIC/42147648-5adf-788d-6387-6706b233ef82}}) Cannot delete file [vsanDatastore] VIC/42147648-5adf-788d-6387-6706b233ef82}) 
INFO[2016-07-22T07:02:04Z] Removing appliance VM network devices        
INFO[2016-07-22T07:02:06Z] Removing VCH vSphere extension               
INFO[2016-07-22T07:02:11Z] Removing Resource Pool "test6"               
INFO[2016-07-22T07:02:11Z] Completed successfully                       

```

### PR DESCRIPTION
**Pull request by emlin**
Fixes #1623 

  Sometimes, VSAN delete parent folder will fail, so if it is VSAN clean up files first, and then delete parent directories.

### CODE REVIEW

**Comment by mdubya66** [comment]
Currently not a stopper for Onsite since we don't have vSan there. Do not merge until we open up

File: lib/install/management/store_files.go [hunk 1]
```diff
@@ -106,16 +106,62 @@ func (d *Dispatcher) deleteDatastoreFiles(ds *object.Datastore, path string, for
...
+}
+
+func (d *Dispatcher) isVSAN(ds *object.Datastore) bool {
+	dsType, _ := ds.Type(d.ctx)
+
+	return dsType == types.HostFileSystemVolumeFileSystemTypeVsan
 }
 
-func (d *Dispatcher) deleteVMFSFiles(ds *object.Datastore, dsPath string, force bool) error {
+func (d *Dispatcher) deleteFilesRecursively(m *object.FileManager, ds *object.Datastore, dsPath string) error {
```
Comment (fdawg4l) [inline lib/install/management/store_files.go]: If I'm reading this routine correctly, it looks in a given directory, checks if it has subfolders, then deletes those subfolders iteratively.   That's not the same as a recursive delete.  Just FYI.

But if this works with actual VMDKs, i'm fine with the implementation.

Comment (emlin) [inline lib/install/management/store_files.go]: Unfortunately, it does not work with non VSAN. So I removed the additional attempt if it failed in normal datastore.

**Comment by mdubya66** [comment]
is there a test for this scenario?

**Comment by mdubya66** [comment]
approved for onsite. please merge as soon as possible.

lgtm

**Comment by dougm** [comment]
lgtm

**Pull request merged**